apps/chat/src/types/storage.ts

```diff
@@ -118,4 +119,18 @@ export interface DialStorage {
   deletePrompt(info: PromptInfo): Observable<void>;
 
   setPrompts(prompts: Prompt[]): Observable<PromptInfo>;
+
+  move(data: MoveModel): Observable<MoveModel>;
+
+  createApplication(
+    application: CustomApplicationModel,
+  ): Observable<ApplicationInfo>;
+
+  updateApplication(application: CustomApplicationModel): Observable<void>;
```

Comment:
What do you think about combine all this methods to updateItem, etc. And give a generic type to props?

Reply:
we have 3 absolutely different storages and it's just Facade-pattern